alarms/postgres/alarms.go

```diff
@@ -0,0 +1,339 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package postgres
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"math"
+	"strings"
+	"time"
+
+	"github.com/absmach/magistrala/alarms"
+	"github.com/absmach/supermq/pkg/errors"
+	repoerr "github.com/absmach/supermq/pkg/errors/repository"
+	"github.com/absmach/supermq/pkg/postgres"
+	"github.com/jmoiron/sqlx"
+)
+
+type repository struct {
+	db *sqlx.DB
+}
+
+var _ alarms.Repository = (*repository)(nil)
+
+func NewAlarmsRepo(db *sqlx.DB) alarms.Repository {
+	return &repository{db: db}
+}
+
+func (r *repository) CreateAlarm(ctx context.Context, alarm alarms.Alarm) (alarms.Alarm, error) {
+	query := `INSERT INTO alarms (id, rule_id, message, status, severity, domain_id, assignee_id, metadata, created_by, created_at)
```

Comment:
We also need the update time. Actually, let's have two: `assigned_at` and `resolved_at`. Other updates should not be possible anyway (i.e. severity change or any other change except for change of assignee).

Reply:
Assigning is an update also resolving is an update